### PR TITLE
Add functional tests for init, train step, and inference for every supported released model

### PR DESCRIPTION
**Pull request by pzelasko**
> [!IMPORTANT]  
> The `Update branch` button must only be pressed in very rare occassions.
> An outdated branch is never blocking the merge of a PR.
> Please reach out to the automation team before pressing that button.

# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

**Collection**: all

# Changelog

- Add specific line by line info of high level changes in this PR.

# Usage

- You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"

**Pre checks**:

- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:

- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed.
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information

- Related to # (issue)

### CODE REVIEW

File: scripts/ci/model_registry.json [hunk 1]
```diff
@@ -0,0 +1,479 @@
+[
```
Comment (blisc) [inline scripts/ci/model_registry.json]: Is the expectation for future model owners to add to this file as they release? Seems prone to human error

File: tests/e2e_nightly/test_model_support_nvidia__magpie_tts_multilingual_357m.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    audio_codes_lens = torch.tensor([T_audio], dtype=torch.long, device=d)
+
+    batch = {
+        "text": text,
+        "text_lens": text_lens,
+        "audio_codes": audio_codes,
+        "audio_codes_lens": audio_codes_lens,
+    }
+
+    batch_output = model.process_batch(batch)
```
Comment (blisc) [inline tests/e2e_nightly/test_model_support_nvidia__magpie_tts_multilingual_357m.py:103]: I'm surprised this works without context audio. I'll have to see what we do under the hood.

File: .github/workflows/cicd-main-speech.yml [hunk 3]
```diff
@@ -199,6 +199,247 @@ jobs:
             script: L2_TTS_InferEvaluate_Magpietts_MoE_ZeroShot
           - runner: self-hosted-azure
             script: L2_TTS_InferEvaluatelongform_Magpietts_MoE_ZeroShot
+          # Model support functional tests
+          - runner: self-hosted-azure
```
Comment (chtruong814) [inline .github/workflows/cicd-main-speech.yml:228]: @pzelasko regarding scheduling this, we actually run nightly tests on main branch already.  Looks like one of the CI tests needs to be fixed.
https://github.com/NVIDIA-NeMo/NeMo/actions/runs/22421911989
https://github.com/NVIDIA-NeMo/NeMo/blob/main/.github/workflows/cicd-main.yml#L17

In any case, to separate these out to run nightly only, it may be better to move the tests to a new group called `e2e-nightly` or whatever you want on it and apply a condition like this:
https://github.com/NVIDIA-NeMo/NeMo/blob/main/.github/workflows/cicd-main.yml#L246

The condition could be something like:
```
if: ${{ github.event_name == 'schedule' }}
needs: unit-tests
```

File: scripts/ci/model_registry.json (hunk 1 above)
Comment (pzelasko) [reply scripts/ci/model_registry.json]: yeah, I'll just get rid of this I guess.

**Comment by pzelasko** [comment]
> I'm wondering if there is a way for us to matrix all of these tests together into one file rather than having a lot of loose files. 

That's what I started from -- the "one python file to rule them all" approach is more compact, but good luck with understanding the flow -- to much logic divergence between different collections and models.

It'll also be easier to add new tests if it's creating a new file (copy + modify) than modify a single complex script.

**Comment by pzelasko** [comment]
Before we merge - my plan is to first ensure every test works, and then move them to an e2e_nightly group like Charlie suggested.

File: .github/workflows/cicd-main-speech.yml (hunk 3 above)
Comment (pzelasko) [reply .github/workflows/cicd-main-speech.yml:228]: @chtruong814 added, can you check if it looks OK? Thanks!
